src/test/java/org/myrobotlab/service/SolrTest.java

```diff
@@ -123,9 +128,42 @@ public void testService() throws Exception {
     query = new SolrQuery("username:joe");
     resp = solr.search(query);
     Assert.assertEquals(1, resp.getResults().getNumFound());
+    
+    
+    // let's search for our vector
+    ArrayList<Float> v = makeVector(384);
+   String queryVec = vecToString(v);
+    
+    
+    query = new SolrQuery("{!knn f=vector topK=10}"+queryVec);
+    resp = solr.search(query);
+    
+    // System.out.println(resp);
+    Assert.assertEquals(1, resp.getResults().getNumFound());
+
+  }
 
+  private String vecToString(ArrayList<Float> v) {

```

Comment:
This looks to me like it's reimplementing the standard `List.toString()`. Any particular reason for it, other than ensuring we aren't reliant on potentially overridden `toString()` implementations?

Reply:
This is just the unit test to create the proper syntax.  We could create a vector class to hold these embeddings.. but ultimately it's a float array.  

The goal of this PR was really just the upgrade and a small example of doing a knn query.